### PR TITLE
Upload Nightly Builds

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
Upload nightly builds to s3

# Description of changes
Add git hub action to upload binaries to s3 nightly

# Tests
Github action set up on my fork

### CODE REVIEW

**Comment by jhnlsn** [comment]
I thought -failfast would fail the build when the first test fails.  It seems to be running all tests even if one fails, making it really difficult to find the FAIL.

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          go-version: ^1.17
+        id: go
+
+      - name: Install zip
+        run: sudo apt install zip
+
+      - name: Install rpm
+        run: sudo apt install rpm
+
+      # Must clone a fresh repo because of how the .git folder is used in the versioning of the agent for packaging
```
Comment (jhnlsn) [inline .github/workflows/nightly-build.yml:33]: I'm not exactly sure what your trying to fix, but the way I understand it, github created this: https://github.com/actions/checkout

**Comment by jhnlsn** [comment]
First of all, this is awesome!  Glad we are taking this direction!

I've seen a lot of projects that release nightly builds, most recently I saw that telegraf releases nightlies: https://github.com/influxdata/telegraf/blob/master/docs/NIGHTLIES.md

- [ ] A couple of things need to be decided before we roll out something like where we are going to store these builds.
- [ ] A readme page that fully outlines that nightly builds are not production ready and should be used at own risk
- [ ] Break out artifacts into their own consumables.  Separate files for downloading deb/rpm/raw binaries.
- [ ] Use `tar.gz` instead of `zip` for everything but windows artifact :)

File: .github/workflows/nightly-build.yml (hunk 1 above)
Comment (sethAmazon) [reply .github/workflows/nightly-build.yml:24]: It has to do with how git tags work.  When using checkout we lose these tags thus we need to clone a fresh repo. What you have posted will not work. https://github.com/actions/checkout explains by default checkout uses the event sha

File: .github/workflows/nightly-build.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+        id: go
+
+      - name: Install rpm
+        run: sudo apt install rpm
```
Comment (adam-mateen) [inline .github/workflows/nightly-build.yml:22]: I suggest adding a comment that `sudo apt install rpm` is done so the Ubuntu host can create both the DEB and RPM install packages.

Also, can you confirm that the RPM that gets built by this Ubuntu machine can be installed on a AL2 (or RedHat or CentOS) host?

**Review by adam-mateen — DISMISSED** [review]

**Pull request merged**